xcat3/plugins/control/openbmc.py

```diff
@@ -0,0 +1,94 @@
+from oslo_log import log
+from xcat3.plugins.control import base
+from xcat3.common import exception
+from xcat3.common import states
+from xcat3.common import rest
+
+import json
+import pdb
+
+LOG = log.getLogger(__name__)
+
+url_dict  = {"set_power_state" : "/xyz/openbmc_project/state/host0/attr/RequestedHostTransition",
+             "get_power_state"  : "/xyz/openbmc_project/state/host0" }
+
+data_dict = {states.POWER_COMMAND_ON   : "xyz.openbmc_project.State.Host.Transition.On",
+             states.POWER_COMMAND_OFF  : "xyz.openbmc_project.State.Host.Transition.Off",
+             states.POWER_COMMAND_RESET: "xyz.openbmc_project.State.Host.Transition.Reboot"}
+
+class OPENBMCPlugin(base.ControlInterface):
+
+    def validate(self, node):
+        bmc = node.control_info.get('bmc')
+        bmcusername = node.control_info.get('bmcusername')
+        bmcpassword = node.control_info.get('bmcpassword')
+        if not bmc:
+            raise exception.MissingParameterValue(
+                _("OPENBMC address was not specified."))
+        if not bmcusername:
+            raise exception.MissingParameterValue(
+                _("OPENBMC username was not specified."))
+        if not bmcpassword:
+            raise exception.MissingParameterValue(
+                _("OPENBMC password was not specified."))
+        return
+
+    def login(self, node):
+        login_url = 'https://' + node.control_info['bmc'] + '/login'
+        login_data = { "data": [ node.control_info['bmcusername'], node.control_info['bmcpassword'] ] }
+        session = rest.login(login_url, login_data)   
+
+        return session
+
+    def power_state_request(self, node, session):
```

Comment:
For ``boot`` command, the flow is ``login->state->on/reset``.
For ``state`` command, the flow is ``login->state``.
And must get session from login, so add a func to avoid login twice.